diff --git a/CMakeLists.txt b/CMakeLists.txt
index d91e3a6c3..84c7d038f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -137,8 +137,10 @@ if(PNG_BUILD_ZLIB)
   endif()
 endif()
 
-# Find the zlib library.
-find_package(ZLIB REQUIRED)
+if (NOT TARGET zlibstatic)
+  # Find the zlib library.
+  find_package(ZLIB REQUIRED)
+endif()
 set(PNG_LINK_LIBRARIES ZLIB::ZLIB)
 
 # Find the math library (unless we already know it's not available or
