diff --git a/CMakeLists.txt b/CMakeLists.txt
index d91e3a6c3..81d41c0e2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -137,9 +137,11 @@ if(PNG_BUILD_ZLIB)
   endif()
 endif()
 
-# Find the zlib library.
-find_package(ZLIB REQUIRED)
-set(PNG_LINK_LIBRARIES ZLIB::ZLIB)
+if (NOT TARGET zlibstatic)
+  # Find the zlib library.
+  find_package(ZLIB REQUIRED)
+  set(PNG_LINK_LIBRARIES ZLIB::ZLIB)
+endif()
 
 # Find the math library (unless we already know it's not available or
 # not needed).
@@ -713,6 +715,8 @@ if(PNG_STATIC)
                              PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>")
   target_include_directories(png_static
                              PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>")
+  target_include_directories(png_static
+                             PUBLIC "${ZLIB_SOURCE_DIR}")
   target_include_directories(png_static
                              SYSTEM
                              INTERFACE "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/libpng${PNGLIB_ABI_VERSION}>")
